sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -295,6 +298,31 @@ object DecimalLiteral {
   def smallerThanSmallestLong(v: Decimal): Boolean = v < Decimal(Long.MinValue)
 }
 
+object LiteralTreeBits {
+  // Singleton tree pattern BitSet for all Literals that are not true, false, or null.
+  val literalBits: BitSet = {
```

Comment:
Constructing BitSets like this is a bit cumbersome. Can we have a helper function to create them?
```scala
def bitSet(bits: Int*): BitSet = {
  val set: BitSet = new BitSet(TreePattern.maxId)
  bits.foreach(set.set)
  bits
}

Reply:
Done. 
Use ImmutableBitSet now - 1 LOC.